src/Kernel/CompiledMethod.class.st

```diff
@@ -1161,7 +1167,7 @@ CompiledMethod >> usesUndeclareds [
 
 { #category : #evaluating }
 CompiledMethod >> valueWithReceiver: aReceiver arguments: anArray [ 
-	^ aReceiver withArgs: anArray executeMethod: self
+	^self receiver: aReceiver withArguments: anArray executeMethod: self
```

Comment:
What about using the MirrorPrimitive class here?
```suggestion
	^MirrorPrimitives withReceiver: aReceiver andArguments: anArray execute: self
```

Reply:
the problem is that this kernel functionality, but MirrorPrimitives is not in the Kernel.

i wanted to do a pass over MirrorPrimitives and use them everywhere, but that should be done as a project in itself.